app/models/concerns/orchestration/dns.rb

```diff
@@ -42,7 +42,7 @@ def rebuild_dns
     results = {}
 
     DnsInterface::RECORD_TYPES.each do |record_type|
-      del_dns_record_safe(record_type)
+      dns_record(record_type).nil? || dns_record(record_type).valid? || del_dns_record_safe(record_type)
```

Comment:
@dLobatog  wouldn't that obfuscate the meaning of del_dns_record_safe? It'd rather add a del_incorrect_dns_record